### PR TITLE
Block promptUISrc from loading iframe with amp-story-consent

### LINKED ISSUE #29154: Block promptUISrc from loading iframe with amp-story-consent
If `promptUISrc` is defined in `amp-consent` config, but `promptUi` is mistakenly omitted, the `promptUiSrc` will still load (but not be shown).

We should prevent this from happening.

### PR DESCRIPTION
**Pull request by micajuine-ho**
Closes #29154

Check if amp-story-consent is a child of the amp-consent tag. If so, and `promptUiSrc` is the only defined field, don't load anything.

### CODE REVIEW

File: extensions/amp-consent/0.1/consent-ui.js [hunk 1]
```diff
@@ -207,7 +214,7 @@ export class ConsentUI {
         );
       }
       this.ui_ = dev().assertElement(promptElement);
-    } else if (promptUISrc) {
+    } else if (promptUISrc && !isStoryConsent) {
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/consent-ui.js:217]: I think it's better to apply the check at a much earlier stage maybe `validateAndParseConfig_()`.

Up to the story team, but I think we should treat this as a user error and use `userAssert()`

File: extensions/amp-consent/0.1/consent-config.js [hunk 2]
```diff
@@ -244,6 +246,25 @@ export class ConsentConfig {
...
+   * @param {!JsonObject} config
+   * @return {!JsonObject}
+   */
+  checkStoryConsent_(config) {
+    const isStoryConsent = !!childElementByTag(
+      this.element_,
+      AMP_STORY_CONSENT_TAG
+    );
+    userAssert(
+      !config.promptUISrc || (config.promptUISrc && !isStoryConsent),
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/consent-config.js:260]: The config comes from an inline json object, use quoted string to refer to property. https://developers.google.com/closure/compiler/docs/limitations#implications-of-global-variable,-function,-and-property-renaming:

File: extensions/amp-consent/0.1/consent-config.js [hunk 3]
```diff
@@ -244,6 +246,25 @@ export class ConsentConfig {
...
+   * Validate if story consent then no promptUiSrc
+   * @param {!JsonObject} config
+   * @return {!JsonObject}
+   */
+  checkStoryConsent_(config) {
+    const isStoryConsent = !!childElementByTag(
+      this.element_,
+      AMP_STORY_CONSENT_TAG
+    );
+    userAssert(
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/consent-config.js:259]: kinda hard to follow here. what about
```
if (childElementByTag(...)) {
  userAssert(!config['promptUISrc'],...)
}
return config
```

File: extensions/amp-consent/0.1/consent-config.js (hunk 2 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/consent-config.js:260]: Thanks for the reference. Will be sure to use this in the future.

File: extensions/amp-consent/0.1/consent-config.js [hunk 4]
```diff
@@ -244,6 +246,26 @@ export class ConsentConfig {
...
+   * @param {!JsonObject} config
+   * @return {!JsonObject}
+   */
+  checkStoryConsent_(config) {
+    if (childElementByTag(
+      this.element_,
+      AMP_STORY_CONSENT_TAG
+    )) {
+      userAssert(
+        !config["promptUISrc"],
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/consent-config.js:260]: nit: I believe it's exactly the same, but please use single quotes for style consistency : )
